Result/ResultType.swift

```diff
@@ -51,6 +51,18 @@ public extension ResultType {
 			ifFailure: Result<U, Error>.Failure)
 	}
 	
+	/// Returns a new Result by mapping `Success`es’ values using `transform`, or wrapping errors thrown as `Failure`s’.
+	public func tryMap<U>(@noescape transform: Value throws -> U) -> Result<U, Error> {
+        return flatMap { value in
+            do {
+                return .Success(try transform(value))
+            }
+            catch {
```

Comment:
For consistency I'd say that `catch` should be on the same line as the closing brace of `do`.